src/prlib/prlib.cpp

```diff
@@ -68,91 +91,162 @@ INCLUDE_ASM(const s32, "prlib/prlib", PrCleanupCamera);
+PR_EXTERN int PrGetAnimationUserData(PR_ANIMATIONHANDLE animation)
+{
+    return ((SpaFileHeader*)animation)->m_userData;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetCameraUserData);
+PR_EXTERN int PrGetCameraUserData(PR_CAMERAHANDLE camera)
+{
+    return ((SpcFileHeader*)camera)->m_userData;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrLinkAnimation);
+PR_EXTERN void PrLinkAnimation(PR_MODELHANDLE model, PR_ANIMATIONHANDLE animation)
+{
+    ((PrModelObject*)model)->LinkAnimation((SpaFileHeader*)animation);
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrUnlinkAnimation);
+PR_EXTERN void PrUnlinkAnimation(PR_MODELHANDLE model)
+{
+    ((PrModelObject*)model)->LinkAnimation(NULL);
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetLinkedAnimation);
+PR_EXTERN PR_ANIMATIONHANDLE PrGetLinkedAnimation(PR_MODELHANDLE model)
+{
+    return (PR_ANIMATIONHANDLE)((PrModelObject*)model)->m_linkedAnim;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrLinkPositionAnimation);
+PR_EXTERN void PrLinkPositionAnimation(PR_MODELHANDLE model, PR_ANIMATIONHANDLE animation)
+{
+    ((PrModelObject*)model)->LinkPositionAnimation((SpaFileHeader*)animation);
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrUnlinkPositionAnimation);
+PR_EXTERN void PrUnlinkPositionAnimation(PR_MODELHANDLE model)
+{
+    ((PrModelObject*)model)->LinkPositionAnimation(NULL);
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetLinkedPositionAnimation);
+PR_EXTERN PR_ANIMATIONHANDLE PrGetLinkedPositionAnimation(PR_MODELHANDLE model)
+{
+    return (PR_ANIMATIONHANDLE)((PrModelObject*)model)->m_linkedPosAnim;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrSelectCamera);
+PR_EXTERN void PrSelectCamera(PR_CAMERAHANDLE camera, PR_SCENEHANDLE scene)
+{
+    ((PrSceneObject*)scene)->SelectCamera((SpcFileHeader*)camera);
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetSelectedCamera);
+PR_EXTERN PR_CAMERAHANDLE PrGetSelectedCamera(PR_SCENEHANDLE scene)
+{
+    return (PR_CAMERAHANDLE)((PrSceneObject*)scene)->m_pCurrentCamera;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetCurrentCamera);
+PR_EXTERN void* PrGetCurrentCamera(PR_SCENEHANDLE scene)
+{
+    return ((PrSceneObject*)scene)->GetCurrentCamera();
+}
 
 INCLUDE_ASM(const s32, "prlib/prlib", PrSetDefaultCamera);
 
 INCLUDE_ASM(const s32, "prlib/prlib", PrSetAppropriateDefaultCamera);
 
 INCLUDE_ASM(const s32, "prlib/prlib", PrShowModel);
 
-PR_EXTERN NaMATRIX<float, 4, 4>* PrGetModelMatrix(PrModelObject *model)
+PR_EXTERN float* PrGetModelMatrix(PR_MODELHANDLE model)
 {
-    if ((model->m_posture & 1) == 0)
+    // Check LSB (hide model flag)
+    if ((((PrModelObject*)model)->m_posture & 1) == 0)
         return NULL;
 
-    return &model->m_matrix;
+    return *((PrModelObject*)model)->m_matrix.m;
 }
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrHideModel);
+PR_EXTERN void PrHideModel(PR_MODELHANDLE model)
+{
+    // Clear LSB (hide model flag)
+    ((PrModelObject*)model)->m_posture &= 0xfffffffe;
```

Comment:
```suggestion
    model->m_posture &= ~1;
```
This is removing the model show flag.